### ISSUE TITLE
mc crashes with "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by barryw** [minio/mc#2526]
## Expected behaviour
`mc cp` from s3 -> minio to complete without error. I have some invalid characters in my s3 paths so I wind up deleting the path in S3 and resuming like this: `mc session resume zzEcTeem`

## Actual behaviour
mc crashes with this stack trace

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0xa26826]

goroutine 10 [running]:
github.com/minio/mc/cmd.doCopySession.func1(0xc42002c240, 0xc4202e0800, 0xc42002c180, 0xc42002c120, 0xc4201bc780, 0xc4202e2750, 0xc4202f8060, 0xc80a40, 0xc4202e2990, 0xc83fc0, ...)
	/q/.q/sources/gopath/src/github.com/minio/mc/cmd/cp-main.go:353 +0x1d6
created by github.com/minio/mc/cmd.doCopySession
	/q/.q/sources/gopath/src/github.com/minio/mc/cmd/cp-main.go:323 +0x3c2
```

## Steps to reproduce the behaviour
I'm resuming my mc session with `mc session resume zzEcTeem`

## mc version
- (paste output of `mc version`)
```Version: 2018-08-18T02:13:04Z
Release-tag: RELEASE.2018-08-18T02-13-04Z
Commit-id: 073ed250fea2beec9b2c5534f9c71b5ebaae79df```

## System information
CentOS 7 on AWS

```
[centos@ip-10-200-20-127 ~]$ uname -a
Linux ip-10-200-20-127.us-west-2.compute.internal 3.10.0-862.11.6.el7.x86_64 #1 SMP Tue Aug 14 21:49:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by kannappanr** [comment]
@barryw Thanks for reaching out to us. We will take a look and try to reproduce this error. Can you please give us examples of few unsupported characters in the path that you have on S3?

**Comment by harshavardhana** [comment]
Fixed by #2520

**Comment by kannappanr** [comment]
Closing this as fixed by #2520.

**Comment by barryw** [comment]
Thanks, @harshavardhana ! Much appreciated!
